applications/HDF5Application/python_scripts/core/file_io.py

```diff
@@ -129,16 +129,16 @@ def __init__(self, settings):
 
     def Get(self, model_part=None):
         file_name = self.filename_getter.Get(model_part)
-        self._InitializeDirectory(file_name)
+        self._InitializeDirectory(file_name, model_part.GetCommunicator().GetDataCommunicator())
         return file_name
 
     @staticmethod
-    def _InitializeDirectory(file_name):
+    def _InitializeDirectory(file_name, data_comm):
         dirname = os.path.dirname(file_name)
         if dirname != '' and not os.path.exists(dirname):
-            if KratosMultiphysics.DataCommunicator.GetDefault().Rank() == 0:
+            if data_comm.Rank() == 0:
                 os.makedirs(dirname)
-            KratosMultiphysics.DataCommunicator.GetDefault().Barrier()
+            data_comm.Barrier()
```

Comment:
Tests fail with this as there are cases where there is no model_part entering here. Before trying to sneak in always the model_part here, in https://github.com/KratosMultiphysics/Kratos/pull/8612#issue-613496700 you mention that 

> 
> The preferred way is to use the `DataCommunicator` of the `ModelPart`, alternatively in special cases it could be accessed by name.
> 
What do you mean that it can be accessed by name? And does it apply here (where sometime there is not a model_part)?

Reply:
I meant that one could ask the parallelEnvironment for the DataComm by name, but this is only for special cases

For HDF I thought it should be relatively straight forward as we always work with the ModelPart. Probably it has to be passed around to some more places.

Regarding the failing tests I am guessing that the mocking causes the issues?

If it is not clear I can have a more detailed look

Reply:
I added `data_comm` as an extra param required for this class. Some test do not require a model_part so Im using the `Testing.GetDefaultDataCommunicator()` there